src/pages/index.js

```diff
@@ -59,19 +181,60 @@ const newCardPopup = new PopupWithForm({
 });
 newCardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
 
 const profileEditPopup = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
   handleFormSubmit: handleEditProfileSubmit,
 });
 
+const avatarEditModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: handleAvatarSubmit,
+});
+
+//Profile creation and editing
+
+const profileAvatar = document.querySelector(".profile__image-container");
+
+profileAvatar.addEventListener("click", () => avatarEditModal.open());
+
+avatarEditModal.setEventListeners();
+
+function handleAvatarSubmit({ link }) {
+  avatarEditModal.renderLoading(true);
+
+  api
+    .updateAvatar(link)
+    .then(() => {
+      userInfo.setAvatar(link);
+      avatarEditModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      avatarEditModal.renderLoading(false);
+    });
+}
+
+//FORM VALIDATION
 const editFormValidator = new FormValidator(
   validationSettings,
   editFormElemenet
 );
 const addFormValidator = new FormValidator(validationSettings, addFormElemenet);
 
+const editAvatarFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```